src/components/AddFundsModal/AddFundsModal.js

```diff
@@ -102,31 +67,15 @@ const AddFundsModal = ({ token, receiveAddress }: Props) => {
       iconName: 'qrDetailed',
       onPress: openReceiveModal,
     },
-    {
+    isExchangeAvailable && {
       key: 'exchange',
       label: t('button.exchange'),
       iconName: 'flip',
       onPress: () => navigation.navigate(EXCHANGE, token && { toAssetCode: token }),
-      hide: hideExchangeOption,
```

Comment:
Can we move this logic outside of array? In case this is falsy options end up having false and then we have to call another method to clean it which for me seems extra load 🤔

Reply:
Falsy arrays seem to be quite established concept in React/RN space, e.g. `View.style` prop can accept array of falsy values to allow for more fluent conditional prop passing.

Reply:
Yeah, but this is different scope as it would break if filters no applied  👀

Reply:
Yeah, but receiving component has proper typing and applies the necessary filtering. Since this uses strict typing and prop is passed directly to related component the risk of breaking should be very low.